python/mleap/pyspark/__init__.py

```diff
@@ -1,5 +1,13 @@
+import pyspark.ml
 import mleap.pyspark
+import mleap.pyspark.feature
+from mleap.pyspark.feature.string_map import StringMap
 import sys
 
 sys.modules['pyspark.ml.mleap'] = mleap
 sys.modules['pyspark.ml.mleap.pyspark'] = sys.modules['mleap.pyspark']
+sys.modules['pyspark.ml.mleap.feature'] = sys.modules['mleap.pyspark.feature']
```

Comment:
Yes, tests.pyspark.string_map_test.StringMapTest.test_serialize_to_bundle fails if this is removed:

```
Error
Traceback (most recent call last):
  File "/Users/juhoautio/.pyenv/versions/2.7.17/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/Users/juhoautio/ideaprojects/mleap/python/tests/pyspark/string_map_test.py", line 63, in test_serialize_to_bundle
    deserialized_pipeline = _deserialize_from_file(pipeline_file)
  File "/Users/juhoautio/ideaprojects/mleap/python/tests/pyspark/string_map_test.py", line 103, in _deserialize_from_file
    return SimpleSparkSerializer().deserializeFromBundle(_to_file_path(path))
  File "/Users/juhoautio/ideaprojects/mleap/python/mleap/pyspark/spark_support.py", line 45, in deserializeFromBundle
    return JavaTransformer._from_java( self._java_obj.deserializeFromBundle(path))
  File "/Users/juhoautio/ideaprojects/mleap/python/.tox/py27/lib/python2.7/site-packages/pyspark/ml/wrapper.py", line 240, in _from_java
    py_stage = py_type._from_java(java_stage)
  File "/Users/juhoautio/ideaprojects/mleap/python/.tox/py27/lib/python2.7/site-packages/pyspark/ml/pipeline.py", line 299, in _from_java
    py_stages = [JavaParams._from_java(s) for s in java_stage.stages()]
  File "/Users/juhoautio/ideaprojects/mleap/python/.tox/py27/lib/python2.7/site-packages/pyspark/ml/wrapper.py", line 227, in _from_java
    py_type = __get_class(stage_name)
  File "/Users/juhoautio/ideaprojects/mleap/python/.tox/py27/lib/python2.7/site-packages/pyspark/ml/wrapper.py", line 221, in __get_class
    m = __import__(module)
ImportError: No module named feature
```

Reply:
instead of using sys.modules to monkeypatch pyspark.ml.mleap, can you use the approach here:
`https://github.com/combust/mleap/blob/master/python/mleap/pyspark/spark_support.py#L32`
^This approach is used widely in mleap, and a lot less unwieldy than `sys.modules[..]=...` imo

Reply:
@talalryz thanks, but isn't this different than the sklearn case because there sklearn is patched with setattr, while here the target for patching is pyspark?

Maybe there is some nicer way around this, but the problem here is that pyspark rewrites /maps the module name, and I suppose that doesn't happen in sklearn:
https://github.com/apache/spark/blob/master/python/pyspark/ml/wrapper.py#L243

I was trying to use setattr instead but this was the "best" I could get to work so far:

```python
import pyspark.ml
import mleap.pyspark
import mleap.pyspark.feature
from mleap.pyspark.feature.string_map import StringMap
import sys

sys.modules['pyspark.ml.mleap'] = mleap
sys.modules['pyspark.ml.mleap.pyspark'] = pyspark
sys.modules['pyspark.ml.mleap.feature'] = feature

setattr(pyspark.ml, 'mleap', mleap)
setattr(pyspark.ml.mleap, 'feature', feature)
setattr(pyspark.ml.mleap.feature, 'StringMap', StringMap)
```

Same number of lines but now mixing both sys.modules & setattr – doesn't seem worth changing to that at least.